### PR TITLE
[padring] Scan role parameters

### PR DESCRIPTION
**Pull request by msfschaffner**
Ideally, the scan role of the pads (`None`, `ScanIn`, `ScanOut`) would be captured as part of the pinout Hjson description.
However, due to the need to keep this information in the foundry repo, this approach with a separate package is taken.

The open-source version is a generic assignment of scan roles (basically all are set to `None`).
The scan role parameters for the synthesized ASIC target are defined in the ASIC repo and will be read in when the "fileset_partner" flag is defined in the build flow.

Note that this is dependent on https://github.com/lowRISC/opentitan/pull/6017, so only the last commit is relevant here.

### CODE REVIEW

File: hw/top_earlgrey/chip_earlgrey_asic.core [hunk 1]
```diff
@@ -12,6 +12,8 @@ filesets:
       - lowrisc:systems:padring
       - "!fileset_partner ? (lowrisc:systems:ast)"
       - "fileset_partner ? (partner:systems:ast)"
+      - "!fileset_partner ? (lowrisc:systems:scan_role_pkg)"
```
Comment (tjaychen) [inline hw/top_earlgrey/chip_earlgrey_asic.core:15]: minor suggestion, could we group the dependencies such that..

```
"fileset_partner ? ..."
"fileset_partner ? ..."
"!fileset_partner ? ..."
"!fileset_partner ? ..."

**Review by tjaychen — APPROVED** [review]
i like this approach! thanks!

**Comment by tjaychen** [comment]
btw, side question... do you think these kinds of scenarios will still be covered by Phil's aliasing proposal?

File: hw/top_earlgrey/chip_earlgrey_asic.core (hunk 1 above)
Comment (msfschaffner) [reply hw/top_earlgrey/chip_earlgrey_asic.core:15]: ah yeah, let's do that.

**Comment by msfschaffner** [comment]
> btw, side question... do you think these kinds of scenarios will still be covered by Phil's aliasing proposal?
> Or would we still need the `fileset_partner` switch long term?

The aliasing proposal from Phil will probably work for this as well, yes.

**Comment by msfschaffner** [comment]
Rebased onto https://github.com/lowRISC/opentitan/pull/6017

**Pull request merged**